src/one_dragon/base/operation/application/plugin_import_service.py

```
@@ -0,0 +1,494 @@
+        file_list = zf.namelist()
+
+        # 检查 zip 文件结构
+        # 情况1: 所有文件都在一个同名子目录下（例如 my_plugin/xxx.py）
+        # 情况2: 文件直接在根目录（例如 xxx_factory.py）
+        has_root_dir = all(
+            name.startswith(plugin_dir_name + '/') or name == plugin_dir_name + '/'
+            for name in file_list if name
+        )
+
+        if has_root_dir:
+            # 直接解压，zip 内部已经有目录结构
```

Ruff (0.15.13)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)